### ISSUE TITLE
[JENKINS-42069] Upgrade plugins without restart

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#10360]
<p>For an upgraded plugin to take effect Jenkins must be restarted. In our environment we can afford a restart only once a week causing unwanted delays. <br/>
Is there a way in which we could upgrade plugins without restarting Jenkins or just reload the plugins(e.g. Reload from disk)? <br/>
Or is there a plan to implement such in the near future?</p>

---
<details><summary><i>Originally reported by <a title="Valerie Fernandes" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=valeriefdes">valeriefdes</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-42069" target="_blank">Upgrade plugins without restart</a></i></summary>
<i><ul>
<li><b>reported by</b>: valeriefdes
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>resolution</b>: Duplicate
<li><b>resolved</b>: 2017-02-15T20:03:58+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>There should be an issue tracking this request, just cannot find it right now.</p>

<p>I doubt this is possible near term.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=zhaoying818">zhaoying818</a>:</i>
<p>is there a plan to implement such in the near future？<img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hadjiyski">hadjiyski</a>:</i>
<p>same for me.<br/>
Wasn't be enough to copy the .hpi plugin under plugins folder (under .jpi extension) and to expand it in the existing subfolder ?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ivan_leon">ivan_leon</a>:</i>
<p>For anybody looking for an update about this: duplicate of <a href="https://issues.jenkins-ci.org/browse/JENKINS-54995" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-54995</a></p>
